### PR TITLE
feat: watch the crd and owns object after crd ready

### PR DESCRIPTION
**Pull request by heww**
In the first time harbor operator running, the `postgresql` and `redisfailover` CRDs are not created at the harbor operator manager bootstrap stage, harbor cluster controller will be failed when it owns the postgresql and redisfailover. The controller can't get the notification from the status changing of `postgresql` and `redisfailover`.

This PR will watch the CRD and try to own the objects again after a CRD ready.

Signed-off-by: He Weiwei <hweiwei@vmware.com>

### CODE REVIEW

**Review by Ghostbaby — APPROVED** [review]

File: pkg/builder/builder.go [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+}
+
+func (blder *Builder) WithOptions(options controller.Options) *Builder {
+	blder.blder.WithOptions(options)
+
+	return blder
+}
+
+// TryOwns owns the object when the crdDependency found, otherwise try owns it after the crdDependency found.
+func (blder *Builder) TryOwns(object runtime.Object, crdDependency string, predicates ...predicate.Predicate) *Builder {
```
Comment (wangcanfengxs) [inline pkg/builder/builder.go:86]: Suggest changing the name to TryOwnsCRD, because it only works for CRD

**Review by wangcanfengxs — COMMENTED** [review]
LGTM

File: pkg/builder/builder.go (hunk 1 above)
Comment (heww) [reply pkg/builder/builder.go:89]: The best name for this method is `TryOwnsWhenCRDReady` but this is too long.

**Review by chlins — APPROVED** [review]
LGTM

**Review by steven-zou — APPROVED** [review]
LGTM

**Review by soulseen — APPROVED** [review]
/lgtm

**Pull request merged**